### PR TITLE
Fix the error reported by OTP and add exception handling to the library

### PR DESCRIPTION
**Pull request by denessapi**

### CODE REVIEW

File: trap/src/main/java/com/cursorinsight/trap/datasource/gesture/TrapPointerCollector.kt [hunk 1]
```diff
@@ -18,9 +18,11 @@ import android.view.MotionEvent.TOOL_TYPE_MOUSE
 import com.cursorinsight.trap.TrapConfig
 import com.cursorinsight.trap.datasource.TrapDatasource
 import com.cursorinsight.trap.datasource.gesture.internal.TrapWindowCallback
+import com.cursorinsight.trap.util.TrapBackgroundExecutor
 import com.cursorinsight.trap.util.TrapTime
 import org.apache.commons.collections4.queue.SynchronizedQueue
 import org.json.JSONArray
+import java.util.Queue
```
Comment (gbence) [inline trap/src/main/java/com/cursorinsight/trap/datasource/gesture/TrapPointerCollector.kt]: Please reorder imports to keep alphabetical order.

File: trap/src/main/java/com/cursorinsight/trap/datasource/gesture/TrapPointerCollector.kt [hunk 2]
```diff
@@ -39,6 +41,7 @@ class TrapPointerCollector(
     @OptIn(ExperimentalStdlibApi::class)
     private val handler = { event: MotionEvent? ->
         if (event != null && event.getToolType(0) == TOOL_TYPE_MOUSE) {
+            val frames = mutableListOf<JSONArray>()
```
Comment (gbence) [inline trap/src/main/java/com/cursorinsight/trap/datasource/gesture/TrapPointerCollector.kt]: Is there a way to manage this in a general way, e.g. in common utility module?

File: trap/src/main/java/com/cursorinsight/trap/TrapManager.kt [hunk 3]
```diff
@@ -113,35 +122,67 @@ class TrapManager internal constructor(
...
-            collector.start(activity)
+        try {
+            reporter.start()
+            for (collector in collectors.values) {
+                collector.start(activity)
+            }
+        } catch (ex: Exception) {
+            Log.e(
+                TrapManager::class.simpleName,
+                "Starting reporter and collectors failed",
```
Comment (gbence) [inline trap/src/main/java/com/cursorinsight/trap/TrapManager.kt]: The subtle difference between the log message above in `run` and this one makes it difficult to find the reason **fast**.

I recommend something like this: "Starting reporter and collectors in runAll() failed."
And above: "Starting reporter and collector in run() failed."

File: trap/src/main/java/com/cursorinsight/trap/datasource/gesture/internal/TrapWindowCallback.kt [hunk 4]
```diff
@@ -53,7 +55,15 @@ class TrapWindowCallback internal constructor(
     }
 
     override fun dispatchTouchEvent(event: MotionEvent?): Boolean {
-        touchHandlers.forEach { it(event) }
+        try {
+            touchHandlers.forEach { it(event) }
+        } catch (ex: Exception) {
+            Log.e(
```
Comment (gbence) [inline trap/src/main/java/com/cursorinsight/trap/datasource/gesture/internal/TrapWindowCallback.kt]: What if there are hundreds of thousands of failiing events?  I recommend adding a failsafe to keep sending failing log messages after some time.

Comment (gbence) [reply trap/src/main/java/com/cursorinsight/trap/datasource/gesture/internal/TrapWindowCallback.kt]: Please check and resolve the same issue below in TrapAccelerometerCollector, TrapGravityCollector, etc.

**Review by gbence — CHANGES_REQUESTED** [review]
There is a typo in one of the commit descriptions:

```
The OS can reuse motion events for performance reasons, if the events
are proessed on a background thread the underlying data might change
    ^^^^^^^^
while the event is processed. This can cause unhandled exceptions or
reporting false data.
```

File: trap/src/main/java/com/cursorinsight/trap/datasource/gesture/TrapPointerCollector.kt (hunk 1 above)
Comment (denessapi) [reply trap/src/main/java/com/cursorinsight/trap/datasource/gesture/TrapPointerCollector.kt]: I've removed it, as it was not needed.

File: trap/src/main/java/com/cursorinsight/trap/datasource/gesture/TrapPointerCollector.kt (hunk 2 above)
Comment (denessapi) [reply trap/src/main/java/com/cursorinsight/trap/datasource/gesture/TrapPointerCollector.kt]: I've refactored this part

File: trap/src/main/java/com/cursorinsight/trap/TrapManager.kt (hunk 3 above)
Comment (denessapi) [reply trap/src/main/java/com/cursorinsight/trap/TrapManager.kt]: I've changed it

File: trap/src/main/java/com/cursorinsight/trap/datasource/gesture/internal/TrapWindowCallback.kt (hunk 4 above)
Comment (denessapi) [reply trap/src/main/java/com/cursorinsight/trap/datasource/gesture/internal/TrapWindowCallback.kt]: I've changed the logging

**Comment by denessapi** [comment]
I've uploaded a new version

File: trap/src/main/java/com/cursorinsight/trap/datasource/gesture/internal/TrapWindowCallback.kt [hunk 5]
```diff
@@ -44,13 +44,16 @@ class TrapWindowCallback internal constructor(
...
             touchHandlers.remove(handler)
         }
-    }
 
-    override fun dispatchTouchEvent(event: MotionEvent?): Boolean {
-        TrapBackgroundExecutor.run {
-            touchHandlers.forEach { it(event) }
+        /**
+         * Removes all touch handlers from the window callback.
+         */fun clear() {
```
Comment (gbence) [inline trap/src/main/java/com/cursorinsight/trap/datasource/gesture/internal/TrapWindowCallback.kt]: Please add a newline here.

Comment (denessapi) [reply trap/src/main/java/com/cursorinsight/trap/datasource/gesture/internal/TrapWindowCallback.kt]: Done

**Review by gbence — APPROVED** [review]

**Pull request merged**